packages/frontend/src/Components/HomePage/index.tsx

```diff
@@ -0,0 +1,176 @@
+
+            if (originalState) { // Se a requisição for retornada do cache
+                // Pegou do cache, faça a requisição novamente para atualizar
+                // os dados e evitar a "viagem no tempo" (conteúdo antigo)
+                // clearCacheByKey(`/Pokemon/getPage/${this.paginationId}`);
+                this.isFetching = false; // Se não a funçaão de fetch não roda
+                const { data } = await this.fetchPokemonsPage(false) || { data: [] };
+                this.setState({ pokemons: [...originalState, ...data] });
+                // Avança para a próxima página
+                const _lastPokemon = data[data.length - 1];
+                this.paginationId = _lastPokemon._id || '';
+            }
+        }).catch(error => {
+            toast.error(error.message);
+            console.log(error);
+            this.isFetching = false;
+        });
+    }
+
+    componentDidMount(): void {
+        this.attachScrollObserver();
+        this.updatePokemonsPage();
+    }
+
+    /** Remove this.scrollObserver do DOM */
+    componentWillUnmount(): void {
+        this.scrollObserverAttached = false;
+        window.removeEventListener("scroll", this.scrollObserver.callback);
+    }
+
+    private suspensePokemons(): React.ReactNode[] {
+        const pokemons: React.ReactNode[] = [];
+        for (let i = 0; i <= 9; i++) {
+            pokemons.push(
+                <PokemonCard key={i} pokemon={{
+                    row: -1, name: 'Carregando...', pokedexId: -1, imageName: 'invalid', generation: 'Carregando...', type1: PokemonType.NORMAL, weather1: PokemonWeather.SUNNY, type2: PokemonType.NORMAL, weather2: PokemonWeather.SUNNY, staminaHP: -1, attack: -1, defense: -1, evolutionState: 'Carregando...', evolved: false, crossGeneration: false, statsSum: -1, legendary: false, acquirable: false, spawns: false, regional: false, raidable: -1, hatchable: -1, shiny: false, nest: false, isNewPokemon: false, notGettable: true, fullCPLevel39: -1, fullCPLevel40: -1, futureEvolve: false,
+                }}
+                    imgSrc="Assets/fa-pokemon-arena.svg"
+                />
+            );
+        }
+        return pokemons;
+    }
+
+    render(): React.ReactNode {
+
+        const { pokemons = null } = this.state;
+        const importSpritesPromise = import("@src/Services/ImportAllPokemonSprites");
+
+        return (<>
+            <Toaster />
+            <main className={classNames(["container", "bg-light", pokemonContainer])} >
+                {pokemons?.length
+                    ? pokemons.map((pokemon) => (
+                        <PokemonCard
+                            pokemon={pokemon}
+                            key={pokemon._id}
+                            imgSrc={`./${pokemon.imageName}.png`}
+                            imgImport
+                            importSpritesPromise={(async () => (importSpritesPromise))()}
```

Comment:
Isso não gera bugs, mas o correto seria
```suggestion
                            importSpritesPromise={(async () => (await importSpritesPromise))()}
```